agent-backend/src/crew/build_crew.py

```diff
@@ -84,6 +84,14 @@ def build_models(self):
 
     def build_tools_and_their_datasources(self):
         for key, tool in self.tools_models.items():
+            
+            model_id = None
+            for _, agent in self.agents_models.items():
+                if tool.id in agent.toolIds:
+                    model_id = agent.modelId
+                    break
+            if model_id is None and self.agents_models:
```

Comment:
A comment here explaining what might cause `model_id` to be `None` at this point would be great!

Reply:
removed